src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,225 @@
+  public ClimberIOTalonFX() {
+    configMotors(leftMotor, rightMotor);
+
+    leftVelocityRPSStatusSignal = leftMotor.getVelocity();
+    leftPositionRotationsStatusSignal = leftMotor.getPosition();
+    leftStatorCurrentAmpsStatusSignal = leftMotor.getStatorCurrent();
+    leftSetpointStatusSignal = leftMotor.getClosedLoopReference();
+
+    rightVelocityRPSStatusSignal = rightMotor.getVelocity();
+    rightPositionMetersStatusSignal = rightMotor.getPosition();
+    rightStatorCurrentAmpsStatusSignal = rightMotor.getStatorCurrent();
+    rightSetpointStatusSignal = rightMotor.getClosedLoopReference();
+
+    leftPositionCurrentRequest = new MotionMagicExpoTorqueCurrentFOC(0.0);
+    rightPositionCurrentRequest = new MotionMagicExpoTorqueCurrentFOC(0.0);
+  }
+
+  /**
+   * Update the inputs based on the current state of the TalonFX leftMotor controller.
+   *
+   * @param inputs the inputs object to update
+   */
+  @Override
+  public void updateInputs(ClimberIOInputs inputs) { // FIXME: May need inputs for both motors
+    BaseStatusSignal.refreshAll(
+        leftVelocityRPSStatusSignal,
+        leftPositionRotationsStatusSignal,
+        leftStatorCurrentAmpsStatusSignal,
+        leftSetpointStatusSignal,
+        leftSupplyCurrentAmpsStatusSignal,
+        rightVelocityRPSStatusSignal,
+        rightPositionMetersStatusSignal,
+        rightStatorCurrentAmpsStatusSignal,
+        rightSetpointStatusSignal,
+        rightSupplyCurrentAmpsStatusSignal);
+
+    inputs.leftVelocityRPS = leftVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.leftPositionMeters =
+        leftPositionRotationsStatusSignal.getValueAsDouble() * DRUM_CIRCUMFERENCE;
+    inputs.leftStatorCurrentAmps = leftStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSupplyCurrentAmps = leftSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSetpoint = leftSetpointStatusSignal.getValueAsDouble();
+
+    inputs.rightVelocityRPS = rightVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.rightPositionMeters = rightPositionMetersStatusSignal.getValueAsDouble();
+    inputs.rightStatorCurrentAmps = rightStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSupplyCurrentAmps = rightSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSetpoint = rightSetpointStatusSignal.getValueAsDouble();
+
+    // inputs.closedLoopError = leftMotor.getClosedLoopError().getValue();
+    // inputs.power =
+    // inputs.controlMode = leftMotor.getControlMode().toString();
+
+    // update configuration if tunables have changed
+    if (kP.hasChanged()
+        || kI.hasChanged()
+        || kD.hasChanged()
+        || kS.hasChanged()
+        || kA.hasChanged()
+        || kV.hasChanged()
+        || kVExpo.hasChanged()
+        || kAExpo.hasChanged()) {
+      TalonFXConfiguration config = new TalonFXConfiguration();
+      MotionMagicConfigs motionMagicConfigs = new MotionMagicConfigs();
```

Comment:
need to get a reference to the MotionMagicConfigs in the TalonFXConfiguration object. For example:
```
motionMagicConfigs = config.MotionMagic
```

Reply:
or... config.MotionMagic = motionMagicConfigs; (not sure which approach is recommended or required)